nexus/src/app/disk.rs

```diff
@@ -496,8 +496,7 @@ impl super::Nexus {
         // (on-disk snapshots, running read-only downstairs) because disks
         // *could* still be using them (if the snapshot has not yet been turned
         // into a regular crucible volume). It will involve some sort of
-        // reference counting for volumes, and probably means this needs to
-        // instead be a saga.
+        // reference counting for volumes.
```

Comment:
This is now a saga, but the comment here seems to suggest becoming a saga would
solve some of the `TODO-correctness` issues above.

However, I'm not sure if this saga does solve those issues.

Reply:
Yeah, I removed the part about saga-ification, but left the TODO for refcounting. I'm not changing anything with respect to refcounting, just making sure that it can maintain crash correctness.

I am not confident enough about the current refcounting scheme to remove the rest of the TODO, tbh.